frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/EditLabels.tsx

```diff
@@ -63,30 +63,37 @@ export function EditLabels(props: { resource?: IResource; displayName?: string;
                                     }
                                     let patch: { op: string; path: string; value?: unknown }[] = []
 
+                                    let deletePatchMap: Map<string, {op: string; path: string; value?: unknown}> = new Map()
+
                                     /* istanbul ignore else */
                                     if (resource!.metadata!.labels) {
-                                        patch = [
-                                            ...patch,
-                                            ...Object.keys(resource!.metadata!.labels).map((key) => {
-                                                key = key.replace(/\//g, '~1')
-                                                return {
-                                                    op: 'remove',
-                                                    path: `/metadata/labels/${key}`,
-                                                }
-                                            }),
-                                        ]
+                                        Object.keys(resource!.metadata!.labels).forEach((key) => {
+                                            key = key.replace(/\//g, '~1')
+                                            deletePatchMap.set(key, {
+                                                op: 'remove',
+                                                path: `/metadata/labels/${key}`,
+                                            })
+                                        })
                                     }
-                                    patch = [
-                                        ...patch,
-                                        ...Object.keys(labels).map((key) => {
-                                            const keyPath = key.replace(/\//g, '~1')
-                                            return {
+
+                                    let addPatchMap: Map<string, {op: string; path: string; value?: unknown}> = new Map()
+                                    Object.keys(labels).forEach((key) => {
+                                        const keyPath = key.replace(/\//g, '~1')
+                                        /* delete from deletePatchMap if still exists */
+                                        if (deletePatchMap.has(keyPath)) {
```

Comment:
and what will happen if the user changes the value of existing key? (by removing it in the UI and adding with a different value?) I think the code should compare the values of the existing key and of the key set in the UI.